### PR TITLE
Fix auto-close tag conditions

### LINKED ISSUE #131: Auto-closing tags issues
Some annoying behaviors around auto-closing tags functionality:
- [x] After writing `/` next to an attribute the auto-closing does not get triggered. Example: `<param name="p"/`
- [x] Writing a `>` immediately after a `<` triggers the auto-closing tag of the parent. Example: `<inputs><>`

### PR DESCRIPTION
**Pull request by davelopez**
Fixes #131

While debugging this issue I found out some of the offsets, after parsing the document, were not matching exactly the `start` and `end` positions on some syntax nodes so I added a bunch of tests to ensure they have the value they are supposed to have.
Some other tests were not very readable and didn't really add much value, so I removed them.

### CODE REVIEW

File: server/galaxyls/services/completion.py [hunk 1]
```diff
@@ -160,8 +161,8 @@ def get_auto_close_tag(self, context: XmlContext, trigger_character: str) -> Opt
         replace_range = None
         is_self_closing = trigger_character == "/"
         if is_self_closing:
-            start = Position(line=context.position.line, character=context.position.character)
-            end_character = context.position.character + 1
+            start = Position(line=context.position.line, character=context.position.character + 1)
```
Comment (bgruening) [inline server/galaxyls/services/completion.py:164]: maybe its good to add a comment here, why the offset is needed

**Review by bgruening — APPROVED** [review]

File: server/galaxyls/services/completion.py (hunk 1 above)
Comment (davelopez) [reply server/galaxyls/services/completion.py:164]: Yeah, this code dealing with offsets is really messy. The +1 is due to the `context.position` been actually set on the `position_before_trigger` character on the previous call but we actually want to replace the trigger character `/` with the snippet `/>$0`.
Thanks for the feedback, I will try to refactor this a bit to see if I can simplify it or even remove the dirty offset math :)

Comment (davelopez) [reply server/galaxyls/services/completion.py:169]: I added some comments to hopefully clarify this dance of character positions :)

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Merci!

**Pull request merged**